pkg/controller/replication_controller.go

```diff
@@ -209,6 +211,14 @@ func (rm *ReplicationManager) getPodControllers(pod *api.Pod) *api.ReplicationCo
 		glog.V(4).Infof("No controllers found for pod %v, replication manager will avoid syncing", pod.Name)
 		return nil
 	}
+	if len(controllers) > 1 {
```

Comment:
I feel like you don't want to rely on behavior in the rc manager to give you deterministic outcomes when rcs shouldn't overlap. I'm not really against sorting, just the people might start relying on it and when we do this right, it'll break.